csrc/python_frontend/python_bindings.cpp

```diff
@@ -2056,14 +2056,19 @@ void initNvFuserPythonBindings(PyObject* module) {
       "__lshift__", "bitwise_left_shift", bitwise_left_shift)
   NVFUSER_PYTHON_BINDING_BINARY_OP_SPECIAL(
       "__rshift__", "bitwise_right_shift", bitwise_right_shift)
-  // In PyTorch, __div__ (//) and __truediv__ (/) are different.
-  // When applied to integer-dtype arguments, they do as expected, returning
-  // integer and float outputs, respectively. When applied to two floating-type
-  // arguments, they return the floor of division for // and plain division for
-  // /. When applied to mixed types, the types are promoted, so the
-  // floating-point behavior is returned.
-  // Our div operator matches the __truediv__ behavior, so we do not implement
-  // __div__.
+  // In python, __truediv__ (/) always returns a float regardless of whether
+  // the input arguments are float or integer. __truediv__ (/) corresponds with
+  // pytorch torch.true_divide(a, b). The __div__ operator is depreciated in
+  // python 3.
+  //
+  // In nvfuser, truediv function has the same semantics as python's
+  // operator __truediv__ (/). The div function truncates the result instead
+  // of promoting it to float, so it has the same semantics as the C++'s (/)
+  // operator. In pytorch, torch.div(a, b, rounding_mode='trunc') corresponds
+  // C-style integer division.
```

Comment:
The first sentence here is incorrect I think. On Python 3.12, `__truediv__` of ints promotes to double first and does no truncation:
```
[ins] In [1]: x = 3

[ins] In [2]: y = 4

[ins] In [3]: x / y
Out[3]: 0.75

[ins] In [4]: x.__truediv__(4)
Out[4]: 0.75
```
To get truncdiv in Python I think you need to manually truncate:
```
[ins] In [17]: ((-30)/4).__trunc__()
Out[17]: -7

[ins] In [18]: (-30)//4
Out[18]: -8
```

Reply:
How is the first sentence incorrect? It is what the comment says in `csrc/ops/arith.h`

> // truediv: promote to float for integer division, has the same semantics as the
> // python's operator /

https://github.com/NVIDIA/Fuser/blob/main/csrc/ops/arith.h#L426-L431

> // div: don't promote to float, instead, truncate the result, this has the same
> // semantics as the C++'s operator /

https://github.com/NVIDIA/Fuser/blob/main/csrc/ops/arith.h#L432-L437

Reply:
I updated the comment to directly mention `csrc/ops/arith.h`.

We are changing the dunder `__truediv__` in our python API to map to c-style division.